packages/backend-storage/src/factory.ts

```diff
@@ -80,5 +80,5 @@ class AmplifyStorageGenerator implements ConstructContainerEntryGenerator {
  */
 export const defineStorage = (
   props: AmplifyStorageProps
-): ConstructFactory<ResourceProvider<StorageResources>> =>
+): ConstructFactory<ResourceProvider<StorageResources> & Construct> =>
```

Comment:
why do we need that `& Construct` here and similar places?

Reply:
Technically we don't. It enforces / allows being able to do something like `Stack.of(backend.storage)` rather than `Stack.of(backend.storage.resources.bucket)`, but maybe that's not needed?

Reply:
The reason I started to wonder was, which construct implementation does implementation of this returns (which is easy to guess for auth, but less obvious for func or storage, for example if we ever get multiple buckets - which one do we return?).
Also returning any construct as `Construct` doesn't seem to offer a lot except resolving stack (as you mentioned) or perhaps retrieving data from cdk context.
I the later example works I'd just remove this and see how this plays out.

Reply:
btw. if accessing stack becomes a pattern we need to simplify perhaps `ResourceProvider` should become `ResourceProvider<Resources & Stack>`